src/Chunk.ts

```diff
@@ -669,11 +669,21 @@ export default class Chunk {
 				chunkSourcemapChain,
 				options.sourcemapExcludeSources!
 			);
-			map.sources = map.sources.map(sourcePath =>
-				normalize(
-					options.sourcemapPathTransform ? options.sourcemapPathTransform(sourcePath) : sourcePath
-				)
-			);
+			map.sources = map.sources.map(sourcePath => {
+				const { sourcemapPathTransform } = options;
+
+				if (sourcemapPathTransform) {
+					const newSourcePath = sourcemapPathTransform(sourcePath);
+
+					if (typeof newSourcePath !== 'string') {
+						throw new Error('sourcemapPathTransform function must return a string.');
```

Comment:
Instead of throwing an error, it would be preferred to use the `error` util and also add a code. As this is a validation, you could do just this: https://github.com/rollup/rollup/blob/35e127c06c4cf86230cb15129f0e3707d15f62f4/src/utils/renderNamePattern.ts#L24-L28

The return is only necessary to make TypeScript happy

Reply:
Ohoo, I was looking for this (and didn't see it ngh) - this is way cleaner :D I'll do that!